wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -0,0 +1,1777 @@
+		 * queries without returning extra tables.
+		 */
+		$this->sqlite_system_tables ['sqlite_sequence']              = 'sqlite_sequence';
+		$this->sqlite_system_tables [ self::DATA_TYPES_CACHE_TABLE ] = self::DATA_TYPES_CACHE_TABLE;
+
+		$this->pdo = $pdo;
+
+		$this->information_schema_builder = new WP_SQLite_Information_Schema_Builder(
+			$this->db_name,
+			array( $this, 'execute_sqlite_query' )
+		);
+		$this->information_schema_builder->ensure_information_schema_tables();
+
+		// Load MySQL grammar.
+		if ( null === self::$grammar ) {
+			self::$grammar = new WP_Parser_Grammar( require self::GRAMMAR_PATH );
+		}
+
+		// Fixes a warning in the site-health screen.
+		$this->client_info = SQLite3::version()['versionString'];
+
+		register_shutdown_function( array( $this, '__destruct' ) );
+
+		// WordPress happens to use no foreign keys.
+		$statement = $this->pdo->query( 'PRAGMA foreign_keys' );
+		// phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
+		if ( $statement->fetchColumn( 0 ) == '0' ) {
+			$this->pdo->query( 'PRAGMA foreign_keys = ON' );
+		}
+		$this->pdo->query( 'PRAGMA encoding="UTF-8";' );
+
+		$valid_journal_modes = array( 'DELETE', 'TRUNCATE', 'PERSIST', 'MEMORY', 'WAL', 'OFF' );
+		if ( defined( 'SQLITE_JOURNAL_MODE' ) && in_array( SQLITE_JOURNAL_MODE, $valid_journal_modes, true ) ) {
+			$this->pdo->query( 'PRAGMA journal_mode = ' . SQLITE_JOURNAL_MODE );
+		}
+	}
+
+	/**
+	 * Destructor
+	 *
+	 * If SQLITE_MEM_DEBUG constant is defined, append information about
+	 * memory usage into database/mem_debug.txt.
+	 *
+	 * This definition is changed since version 1.7.
+	 */
+	public function __destruct() {
+		if ( defined( 'SQLITE_MEM_DEBUG' ) && SQLITE_MEM_DEBUG ) {
+			$max = ini_get( 'memory_limit' );
+			if ( is_null( $max ) ) {
+				$message = sprintf(
+					'[%s] Memory_limit is not set in php.ini file.',
+					gmdate( 'Y-m-d H:i:s', $_SERVER['REQUEST_TIME'] )
+				);
+				error_log( $message );
+				return;
+			}
+			if ( stripos( $max, 'M' ) !== false ) {
+				$max = (int) $max * MB_IN_BYTES;
+			}
+			$peak = memory_get_peak_usage( true );
+			$used = round( (int) $peak / (int) $max * 100, 2 );
+			if ( $used > 90 ) {
+				$message = sprintf(
+					"[%s] Memory peak usage warning: %s %% used. (max: %sM, now: %sM)\n",
+					gmdate( 'Y-m-d H:i:s', $_SERVER['REQUEST_TIME'] ),
+					$used,
+					$max,
+					$peak
+				);
+				error_log( $message );
+			}
+		}
+	}
+
+	/**
+	 * Get the PDO object.
+	 *
+	 * @return PDO
+	 */
+	public function get_pdo() {
+		return $this->pdo;
+	}
+
+	/**
+	 * Method to return inserted row id.
+	 */
+	public function get_insert_id() {
+		return $this->last_insert_id;
+	}
+
+	/**
+	 * Method to return the number of rows affected.
+	 */
+	public function get_affected_rows() {
+		return $this->affected_rows;
+	}
+
+	/**
+	 * Method to execute query().
+	 *
+	 * Divide the query types into seven different ones. That is to say:
```

Comment:
nice comment

Reply:
Even this one is from the legacy translator. So the comment won't be fully up to date, I guess and needs to be improved.